apph-front/src/static/components/SignUp.tsx

```diff
@@ -0,0 +1,180 @@
+    const data = new FormData(event.currentTarget);
+    const email = data.get('email')?.toString();
+    const password = data.get('password')?.toString();
+    const confirmPassword = data.get('confirmPassword')?.toString();
+    const firstName = data.get('firstName')?.toString();
+    const lastName = data.get('lastName')?.toString();
+    if (password != confirmPassword) {
+      setErrorMessage(
+        'Ces mots de passe ne correspondent pas. Veuillez réessayer.'
+      );
+    } else if (email && password && firstName && lastName) {
+      UserService.signUp(
+        email,
+        password,
+        firstName,
+        lastName,
+        () => {
+          console.log('Création de compte réussie');
+          navigate('/signIn');
+        },
+        (errorMessage: string) => {
+          setErrorMessage(errorMessage);
+        }
+      );
+    }
+  };
+
+  return (
+    <Container component="main" maxWidth="xs">
+      <CssBaseline />
+      <Box
+        sx={{
+          marginTop: 8,
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center'
+        }}
+      >
+        <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
+          <LockOutlinedIcon />
+        </Avatar>
+        <Typography component="h1" variant="h5">
+          Créer un compte
+        </Typography>
+        <Box component="form" noValidate onSubmit={handleSubmit} sx={{ mt: 3 }}>
+          <Grid container spacing={2}>
+            <Grid item xs={12} sm={6}>
+              <TextField
+                required
+                fullWidth
+                id="firstName"
+                label="Prénom"
+                name="firstName"
+                autoComplete="given-name"
+                autoFocus
+              />
+            </Grid>
+            <Grid item xs={12} sm={6}>
+              <TextField
+                required
+                fullWidth
+                id="lastName"
+                label="Nom"
+                name="lastName"
+                autoComplete="family-name"
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <TextField
+                required
+                fullWidth
+                id="email"
+                label="Email"
+                name="email"
+                autoComplete="email"
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <TextField
+                required
+                fullWidth
+                id="password"
+                label="Mot de passe"
+                name="password"
+                type="password"
+                autoComplete="new-password"
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <TextField
+                required
+                fullWidth
+                id="confirmPassword"
+                label="confirmer le mot de passe"
+                name="confirmPassword"
+                type="confirmPassword"
```

Comment:
![image](https://user-images.githubusercontent.com/30409365/166420313-cbb7a393-df10-407f-91b2-0d37a637aa66.png)
ça ne marche pas pour cacher le mdp, c'est plutôt un type "password"?
```suggestion
                type="password"
```